test/e2e/crud.go

```diff
@@ -26,38 +26,47 @@ import (
 	. "github.com/onsi/ginkgo"
 )
 
-var _ = Describe("Federated types", func() {
-	f := framework.NewFederationFramework("federated-types")
-
-	fedTypeConfigs := federatedtypes.FederatedTypeConfigs()
-	for kind, _ := range fedTypeConfigs {
-		// Bind the type config inside the loop to ensure the ginkgo
-		// closure gets a different value for every loop iteration.
-		//
-		// Reference: https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables
-		fedTypeConfig := fedTypeConfigs[kind]
-
-		Describe(fmt.Sprintf("%q resources", kind), func() {
-			It("should be created, read, updated and deleted successfully", func() {
-				// Initialize an in-memory controller if configuration requires
-				f.SetUpControllerFixture(kind, fedTypeConfig.AdapterFactory)
-
-				userAgent := fmt.Sprintf("crud-test-%s", kind)
-				adapter := fedTypeConfig.AdapterFactory(f.FedClient(userAgent))
-				namespaceAdapter, ok := adapter.(*federatedtypes.FederatedNamespaceAdapter)
-				if ok {
-					namespaceAdapter.SetKubeClient(f.KubeClient(userAgent))
-				}
-				clusterClients := f.ClusterClients(userAgent)
-				crudTester := common.NewFederatedTypeCrudTester(framework.NewE2ELogger(), adapter, clusterClients, framework.PollInterval, framework.SingleCallTimeout)
-				clusterNames := []string{}
-				for name, _ := range clusterClients {
-					clusterNames = append(clusterNames, name)
-				}
-				template, placement, override := federatedtypes.NewTestObjects(fedTypeConfig.Kind, f.TestNamespaceName(), clusterNames)
-
-				crudTester.CheckLifecycle(template, placement, override)
+func addE2ECrudTests() {
```

Comment:
I think it would be preferable to skip within the It() clause.  Likely worth porting over the [skip code](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L293) from k/k (to test/e2e/framework/util.go).

Reply:
You're right, this makes it much cleaner! But first I had to fix a stale bind issue with the `kind` closure variable.